static/js/markdown.js

```diff
@@ -187,15 +187,18 @@ export function apply_markdown(message) {
             const actual_full_name = helpers.get_actual_name_from_user_id(user_id);
             return `${str}${_.escape(actual_full_name)}</span>`;
         },
-        groupMentionHandler(name) {
+        groupMentionHandler(name, silently) {
             const group = helpers.get_user_group_from_name(name);
             if (group !== undefined) {
-                if (helpers.is_member_of_user_group(group.id, helpers.my_user_id())) {
+                const group_name = (silently ? "" : "@") + group.name;
+                const classes = silently ? "user-group-mention silent" : "user-group-mention";
+                if (helpers.is_member_of_user_group(group.id, helpers.my_user_id()) && !silently) {
                     message.mentioned = true;
                 }
```

Comment:
This is hard to read.  I think it'd be a lot better to just have a single `if (silently)` and set `group_name` or classes within that, than having 3 checks for silently in inline conditionals.

Reply:
I initially did this, but I don't know why I thought I shouldn't expand the code :facepalm: and try to keep it compact.
Thanks! for explaining.